### ISSUE TITLE
bug: with many files, the grid option doesn't work

### ISSUE THREAD

**Issue by kiyoon** [eza-community/eza#799]
- The version of eza being used (`eza --version`): 0.17.2
- The command-line arguments you are using: `eza`, `eza --icons auto` etc.
- Your operating system and hardware platform: Ubuntu 20.04

Without the `-1` option it still prints every file per line. All of the file names are shown as such, with 4 character codes.

![image](https://github.com/eza-community/eza/assets/12980409/0ee1460b-0443-413e-b8a7-ec7f14093ed8)

**Comment by PThorpe92** [comment]
Hey thanks for reporting the issue. A couple questions, 

Could you list out the rest of the exact arguments you are using? 

how many files in a directory is "many", and the same args in a smaller directory is producing different output? 

 >All of the file names are shown as such, with 4 character codes.

Do you mean that those aren't the file names? 


Trying to re-create this issue, so as much info as you can give would be helpful :+1:

**Comment by kiyoon** [comment]
@PThorpe92 No argument was needed to reproduce this. It works like this with whatever I use, with or without `--icons` for example. With `exa` the grid works well.

This was enough to reproduce:

```sh
for a in {0001..0401}; do touch $a; done
```

With 400 files it works as expected, but not with 401 files.

**Comment by PThorpe92** [comment]
Interesting how this doesn't reproduce in gnome. Thanks for the info I'll try to repro this as I use wezterm as well.

**Comment by kiyoon** [comment]
@PThorpe92 My theory is that there exist some edge cases where it doesn't align the grid, and maybe with gnome terminal if I had the exact same column it might happen too.

**Comment by PThorpe92** [comment]
Strange, I expected to be able to reproduce this, as I have seen kind of random errors in the past with grid width. but I tried on Wez, alactritty, kitty, xfce-term and xterm on eza `v0.17.3` and got the proper amt of columns every time, even at different window sizes.
Perhaps try upgrading to`17.3` and see if that fixes it?

**Comment by kiyoon** [comment]
Thanks for the time to try many settings but I'm sorry you couldn't reproduce. I tried with the `v0.17.3` but the issue still exists.

Additional info that may be helpful

```
$ tput cols
166

$ tput lines
34
```

It seems like the number of row doesn't matter, but the width matters. I tried with smaller font size and set the width to 166, and I can still reproduce it.

**Comment by kiyoon** [comment]
OK, with Gnome Terminal it happens exactly the same as Wezterm so it wasn't the issue with the terminal. My bad.

With width=166 you should be able to reproduce it.

**Comment by PThorpe92** [comment]
Yeah after quite a bit of testing, I was able to reproduce it. Super strange, because the command that I use for my alias (`alias ls="eza --icons --color-scale -lAh --git --git-repos --no-permissions --grid"`) will display properly, but `eza --icons` will not.

but `eza --icons -lA --grid` will display properly, however if you take away the -lA, `--icons --grid` will not... 

Very strange indeed, especially because with `cols` at 158, and 176, it works as expected :confused: 

Looking into this :+1:

**Comment by kiyoon** [comment]
That is really weird! But I'm glad you were able to reproduce it and I appreciate it for maintaining the awesome project :smile:

**Comment by tertsdiepraam** [comment]
I'm looking into this, it's a bug in `uutils-term-grid`. I've been able to reproduce it in the test suite there.

**Comment by tertsdiepraam** [comment]
I've fixed this in https://github.com/uutils/uutils-term-grid/pull/34, but we'll need to port `eza` to use the new version of `uutils-term-grid` to use the fix. It was a weird combination of small mistakes in the computation of some heuristics, which is why it was so sensitive to those specific values.

**Comment by kiyoon** [comment]
Thank you so much!

**Comment by kiyoon** [comment]
Closing as it seems to have been fixed.
